packages/fxa-settings/src/pages/Signup/Signup/index.tsx

```diff
@@ -0,0 +1,491 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { useForm } from 'react-hook-form';
+import { logViewEvent, usePageViewEvent } from '../../../lib/metrics';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useState } from 'react';
+import FormPasswordWithBalloons from '../../../components/FormPasswordWithBalloons';
+import Checkbox from '../../../components/Settings/Checkbox';
+import InputText from '../../../components/InputText';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { ReactComponent as IconClose } from 'fxa-react/images/close.svg';
+import { useFtlMsgResolver } from '../../../models';
+
+/**
+ * Fields available in each newsletter:
+ * - `label` is the label shown to the user.
+ * - `slug` is the id passed to Basket
+ */
+type Newsletter = {
+  label: string;
+  slug: string;
+};
+
+/**
+ * Fields available in each engine:
+ * - `checked` whether the item should be checked when CWTS opens.
+ * - `id` of the engine, must be the name the browser uses.
+ * - `text` to display when CWTS opens
+ * - `test` if defined, function used to test whether CWTS is available
+ *    for the given `userAgent`. Should return `true` or `false`.
+ */
+type Engine = {
+  checked: boolean;
+  id: string;
+  text: string;
+  test?: () => boolean;
+};
+interface SharedProps {
+  email: string;
+  // canChangeEmail is true if not from relying party or force_auth
+  canChangeEmail?: boolean;
+  // POCKET_CLIENTIDS.includes(this.relier.get('clientId'))
+  isPocketClient?: boolean;
+}
+
+// CWTS is enabled if relier is sync or multiService, broker is OAuth
+// CWTS and newsletters cannot both be enabled
+type ConditionalProps =
+  | {
+      isCWTSEnabled?: boolean;
+      areNewslettersEnabled?: never;
+    }
+  | {
+      isCWTSEnabled?: never;
+      areNewslettersEnabled?: boolean;
+    }
+  | {
+      isCWTSEnabled?: never;
+      areNewslettersEnabled?: never;
+    };
+
+export type SignupProps = SharedProps & ConditionalProps;
+
+type FormData = {
+  newPassword: string;
+  confirmPassword: string;
+  userAge: string;
+};
+
+const Signup = ({
+  email,
+  canChangeEmail = true,
+  isPocketClient,
+  isCWTSEnabled,
+  areNewslettersEnabled,
+}: SignupProps & RouteComponentProps) => {
+  usePageViewEvent('signup', {
+    entrypoint_variation: 'react',
+  });
+
+  const onFocusMetricsEvent = 'signup.engage';
+
+  const [age, setAge] = useState<number>(0);
+  const [ageCheckErrorText, setAgeCheckErrorText] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [newPasswordErrorText, setNewPasswordErrorText] = useState('');
+  const [
+    isAccountSuggestionBannerVisible,
+    setIsAccountSuggestionBannerVisible,
+  ] = useState<boolean>(isPocketClient ? true : false);
```

Comment:
Q: is `isPocketClient` already a Boolean? Does it need the `? true : false` or is that somehow guarding against possible null/undefined values? (if so, you might also be able to cast w/ `!!isPocketClient`, but 🤷 ).

Reply:
Good call, I think setting a default value in the props would resolve this.